app/models/metric/ci_mixin/capture.rb

```diff
@@ -80,10 +80,12 @@ def perf_capture_queue(interval_name, options = {})
       :state       => ['ready', 'dequeue'],
     }
 
+    messages = MiqQueue.where.not(:method_name => 'perf_capture_realtime').where(queue_item).index_by(&:args)
```

Comment:
`EXPLAIN` plan of both queries seems the same.

Mine:
```
Index Scan using miq_queue_put_idx on miq_queue  (cost=0.41..8.53 rows=1 width=1474)
   Index Cond: (((class_name)::text = 'ManageIQ::Providers::Redhat::InfraManager::Vm'::text) AND ((queue_name)::text = 'redhat'::text) AND ((zone)::text = 'default'::text))
   Filter: (((method_name)::text <> 'perf_capture_realtime'::text) AND ((state)::text = ANY ('{ready,dequeue}'::text[])) AND (instance_id = '1000000003563'::bigint) AND ((role)::text = 'ems_metrics_collector'::text))
(3 rows)
```

Yours
```
 Index Scan using miq_queue_put_idx on miq_queue  (cost=0.41..8.53 rows=1 width=1474)
   Index Cond: (((class_name)::text = 'ManageIQ::Providers::Redhat::InfraManager::Vm'::text) AND ((queue_name)::text = 'redhat'::text) AND ((zone)::text = 'default'::text))
   Filter: (((method_name)::text = ANY ('{perf_capture_historical,perf_capture_hourly}'::text[])) AND ((state)::text = ANY ('{ready,dequeue}'::text[])) AND (instance_id = '1000000003563'::bigint) AND ((role)::text = 'ems_metrics_collector'::text))
(3 rows)
```

What am I missing?